sklearn/model_selection/_validation.py

```diff
@@ -625,8 +625,12 @@ def cross_val_predict(estimator, X, y=None, groups=None, cv=None,
                       pre_dispatch='2*n_jobs', method='predict'):
     """Generate cross-validated estimates for each input data point
 
-    It is not appropriate to pass these predictions into an evaluation
-    metric. Use :func:`cross_validate` to measure generalization error.
+    Using these predictions to measure the generalization performance of the model 
+    can lead to different results than `cross_val` due to the aggregation of the 
+    predictions. 
+    
+    Moreover only metrics compatible with pooling have to be considered and test
```

Comment:
what does pooling mean in this context?

Reply:
From 
Airola, Antti, et al. "A comparison of AUC estimators in small-sample studies." Machine learning in systems biology. 2009.
They differentiate pooling from averaging. In pooling, we aggregate the predictions of the different models and compute one metric. In averaging, we compute one metric for each folds and we compute the average